Pkg 2916 0.18.0

danpetry opened the pull request:
**Destination channel:** defaults

### Links

- [PKG-5219](https://anaconda.atlassian.net/browse/PKG-5219) 
- [Upstream repository](https://github.com/pytorch/vision/tree/v0.18.0)

### Explanation of changes:

- To support pytorch v2.3.0
- Note torchvision v1.18.1 supports v2.3.1
- Pulled conda-forge
- Updated for cuda support, log for cuda 11.8 build [here](https://github.com/user-attachments/files/16199880/2correctselectors.log)
- enable tests
- use ffmpeg again, seems to be working
- don't use libjpeg-turbo as it would require changing to it from libjpeg distro-wide

[PKG-5219]: https://anaconda.atlassian.net/browse/PKG-5219?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

h-vetinari committed:
Merge pull request #25 from h-vetinari/bump

Bump to 0.9.1

h-vetinari committed:
bump build number

h-vetinari committed:
wip

h-vetinari committed:
add upper version bound to pillow due to broken tests

github-actions[bot] committed:
Add pillow upper version bound to unbreak test suite

automerged PR by conda-forge/automerge-action

h-vetinari committed:
argh...

regro-cf-autotick-bot committed:
updated v0.10.0

h-vetinari committed:
remove obsolete patches

h-vetinari committed:
reset build number and make it bot-upgradeable

h-vetinari committed:
update pillow version range

h-vetinari committed:
update pillow version constraint

regro-cf-autotick-bot committed:
Rebuild for pytorch

regro-cf-autotick-bot committed:
MNT: Re-rendered with conda-build 3.21.4, conda-smithy 3.11.0, and conda-forge-pinning 2021.07.29.23.13.45

h-vetinari committed:
remove compatibility output that's not used by upstream anymore

h-vetinari committed:
skip test relying on unavailable nnpack on osx

h-vetinari committed:
pin pillow to <8.3 again

h-vetinari committed:
remove pytorch pin

h-vetinari committed:
Pin variant (cpu/gpu) better

Co-authored-by: Mark Harfouche <mark.harfouche@gmail.com>

hmaarrfk committed:
Merge pull request #28 from regro-cf-autotick-bot/rebuild-pytorch-0-1_h2b1f7e

Rebuild for pytorch

h-vetinari committed:
fix comment for skipped test

h-vetinari committed:
add av as test dependency

hmaarrfk committed:
Update meta.yaml

h-vetinari committed:
add patch for pillow 8.3 compatibility

hmaarrfk committed:
Merge pull request #27 from regro-cf-autotick-bot/0.10.0_h94d557

torchvision v0.10.0

regro-cf-autotick-bot committed:
updated v0.10.1

regro-cf-autotick-bot committed:
MNT: Re-rendered with conda-build 3.21.4, conda-smithy 3.12, and conda-forge-pinning 2021.09.27.04.24.35

h-vetinari committed:
undo unwanted changes by bot

h-vetinari committed:
Merge pull request #30 from regro-cf-autotick-bot/0.10.1_haa9943

torchvision v0.10.1

regro-cf-autotick-bot committed:
updated v0.11.1

regro-cf-autotick-bot committed:
MNT: Re-rendered with conda-build 3.21.4, conda-smithy 3.13.0, and conda-forge-pinning 2021.10.21.11.39.39

hmaarrfk committed:
Update meta.yaml

hmaarrfk committed:
Delete recipe/patches directory

hmaarrfk committed:
Update meta.yaml

hmaarrfk committed:
test in verbose and pin pytorch better

hmaarrfk committed:
Explain wy we are skipping tests

hmaarrfk committed:
lint

hmaarrfk committed:
Add pytest-mock

hmaarrfk committed:
Update meta.yaml

hmaarrfk committed:
Delete the models tests to skip it.

conda-forge-admin committed:
[ci skip] [skip ci] [cf admin skip] ***NO_CI*** admin migration CondaForgeGHAWithMain

regro-cf-autotick-bot committed:
updated v0.11.2

conda-forge-admin committed:
[ci skip] [skip ci] [cf admin skip] ***NO_CI*** admin migration CondaForgeGHAWithMain

h-vetinari committed:
minor cleanup

h-vetinari committed:
MNT: Re-rendered with conda-build 3.21.7, conda-smithy 3.16.0, and conda-forge-pinning 2021.12.20.19.09.43

h-vetinari committed:
build with ffmpeg

h-vetinari committed:
add patch to avoid hard-coded GCC

Commit:
MNT: Re-rendered with conda-build 3.21.7, conda-smithy 3.16.1, and conda-forge-pinning 2022.01.20.12.50.15

hmaarrfk committed:
Justify which tests we are skipping

hmaarrfk committed:
Skip more tests

hmaarrfk committed:
Skip a few more tests specifically for osx

github-actions[bot] committed:
torchvision v0.11.2

automerged PR by conda-forge/automerge-action

hmaarrfk committed:
Run more of the test suite

hmaarrfk committed:
Merge pull request #35 from hmaarrfk/try_testing

Run more of the test suite

hmaarrfk committed:
Cleanup the recipe an skip a flaky test

hmaarrfk committed:
MNT: Re-rendered with conda-build 3.21.7, conda-smithy 3.16.1, and conda-forge-pinning 2022.01.22.21.08.53

hmaarrfk committed:
skip more tests

hmaarrfk committed:
Merge pull request #37 from hmaarrfk/clenaup_recipe

Cleanup recipe

hmaarrfk committed:
Attempt to pin pytorch correctly

hmaarrfk committed:
MNT: Re-rendered with conda-build 3.21.7, conda-smithy 3.16.2, and conda-forge-pinning 2022.01.29.15.10.18

hmaarrfk committed:
Merge pull request #42 from hmaarrfk/pin_correctly

Pin correctly

regro-cf-autotick-bot committed:
updated v0.11.3

github-actions[bot] committed:
torchvision v0.11.3

automerged PR by conda-forge/automerge-action

hmaarrfk committed:
Try to redo pinnings so that we use run exports

hmaarrfk committed:
Update meta.yaml

hmaarrfk committed:
Update meta.yaml

hmaarrfk committed:
Update meta.yaml

hmaarrfk committed:
Merge pull request #44 from hmaarrfk/redo_pinnings

Try to redo pinnings so that we use run exports

h-vetinari committed:
bump to 0.12.0

h-vetinari committed:
remove upstreamed patch

h-vetinari committed:
unpin ffmpeg

h-vetinari committed:
remove redundant line

h-vetinari committed:
remove pyproject.toml

conda-forge-admin committed:
[ci skip] [skip ci] [cf admin skip] ***NO_CI*** turning off CI for master to main migration

conda-forge-admin committed:
[ci skip] [skip ci] [cf admin skip] ***NO_CI*** admin migration CondaForgeMasterToMain

conda-forge-admin committed:
[ci skip] [skip ci] [cf admin skip] ***NO_CI*** turning on CI for master to main migration

regro-cf-autotick-bot committed:
Rebuild for python310

h-vetinari committed:
bump build number

Commit:
MNT: Re-rendered with conda-build 3.21.8, conda-smithy 3.18.0, and conda-forge-pinning 2022.03.21.01.01.44

hmaarrfk committed:
Let it pick up the run_exports from pytorch

h-vetinari committed:
temporary: force solver to pick pytorch build with correct run-export

regro-cf-autotick-bot committed:
Rebuild for pytorch111

regro-cf-autotick-bot committed:
MNT: Re-rendered with conda-build 3.21.8, conda-smithy 3.18.0, and conda-forge-pinning 2022.03.21.04.26.58

h-vetinari committed:
Merge pull request #48 from h-vetinari/rerender

Rebuild to pick up correct pytorch run_export

h-vetinari committed:
Revert "temporary: force solver to pick pytorch build with correct run-export"

This reverts commit 44edce11047cb6d7f72f7d7fb99bedd8d8be2d3a.

Commit:
MNT: Re-rendered with conda-build 3.21.8, conda-smithy 3.18.0, and conda-forge-pinning 2022.03.21.04.26.58

h-vetinari committed:
Merge pull request #49 from regro-cf-autotick-bot/rebuild-pytorch111-0-1_h760108

Rebuild for pytorch111

h-vetinari committed:
add requests as dep

h-vetinari committed:
backport patch for spurious test failures

h-vetinari committed:
Merge pull request #45 from h-vetinari/bump

torchvision 0.12.0

regro-cf-autotick-bot committed:
Rebuild for ffmpeg44

regro-cf-autotick-bot committed:
MNT: Re-rendered with conda-build 3.21.8, conda-smithy 3.19.0, and conda-forge-pinning 2022.03.28.22.11.58

hmaarrfk committed:
Update meta.yaml

github-actions[bot] committed:
Rebuild for ffmpeg44 (#53)

automerged PR by conda-forge/automerge-action

regro-cf-autotick-bot committed:
Rebuild for pytorch112

regro-cf-autotick-bot committed:
MNT: Re-rendered with conda-build 3.21.9, conda-smithy 3.21.0, and conda-forge-pinning 2022.07.18.11.18.34

hmaarrfk committed:
Update to 0.13.0

hmaarrfk committed:
Delete 0001-More-robust-check-in-tests-for-16-bits-images.patch

hmaarrfk committed:
Fixup PTX location at the end of the arch list

hmaarrfk committed:
Skip many more tests that fail on CIs

hmaarrfk committed:
Adjust test skips for CIs

hmaarrfk committed:
Use lgpl ffmpeg

h-vetinari committed:
Build without ffmpeg altogether

Commit:
MNT: Re-rendered with conda-build 3.21.9, conda-smithy 3.21.0, and conda-forge-pinning 2022.07.20.21.54.10

hmaarrfk committed:
Merge pull request #58 from regro-cf-autotick-bot/rebuild-pytorch112-0-1_h34c4cb

Rebuild for pytorch112

conda-forge-curator[bot] committed:
[ci skip] [skip ci] [cf admin skip] ***NO_CI*** admin migration CondaForgeAutomergeUpdate

conda-forge-curator[bot] committed:
[ci skip] [skip ci] [cf admin skip] ***NO_CI*** admin migration DotConda

h-vetinari committed:
bump to 0.14.0

h-vetinari committed:
manually add pytorch 1.13 migrator

h-vetinari committed:
skip CUDA<11.2

h-vetinari committed:
MNT: Re-rendered with conda-build 3.23.3, conda-smithy 3.22.1, and conda-forge-pinning 2022.12.09.20.17.28

hmaarrfk committed:
Merge pull request #62 from h-vetinari/bump

Bump to 0.14

Tobias-Fischer committed:
Add aarch64

Tobias-Fischer committed:
Bump

Commit:
MNT: Re-rendered with conda-build 3.23.3, conda-smithy 3.22.1, and conda-forge-pinning 2023.01.18.19.16.13

Tobias-Fischer committed:
Skip more tests due to incompat with new numpy

Tobias-Fischer committed:
Flaky test

Tobias-Fischer committed:
Try again without test skip to see if failure always happens

Tobias-Fischer committed:
Disable test_video_clips_custom_fps again

See https://github.com/conda-forge/torchvision-feedstock/pull/64#issuecomment-1396753883 for explanation

hmaarrfk committed:
Merge pull request #64 from Tobias-Fischer/patch-1

Add aarch64

regro-cf-autotick-bot committed:
Rebuild for python311

regro-cf-autotick-bot committed:
updated v0.14.1

h-vetinari committed:
remove stuff the bot messed up

github-actions[bot] committed:
torchvision v0.14.1 (#66)

automerged PR by conda-forge/automerge-action

regro-cf-autotick-bot committed:
Rebuild for jpegturbo migration

regro-cf-autotick-bot committed:
MNT: Re-rendered with conda-build 3.23.3, conda-smithy 3.23.0, and conda-forge-pinning 2023.03.08.02.10.48

hmaarrfk committed:
Merge pull request #68 from regro-cf-autotick-bot/rebuild-jpeg_to_libjpeg_turbo-0-3_h7df6a6

Rebuild for jpegturbo migration

Tobias-Fischer committed:
Bump to 0.15.1

Commit:
MNT: Re-rendered with conda-build 3.24.0, conda-smithy 3.23.1, and conda-forge-pinning 2023.03.29.16.02.14

Tobias-Fischer committed:
SHA

regro-cf-autotick-bot committed:
Rebuild for pytorch20

h-vetinari committed:
MNT: Re-rendered with conda-build 3.24.0, conda-smithy 3.23.1, and conda-forge-pinning 2023.04.07.02.53.30

h-vetinari committed:
add references/ folder to test-source_files

Tobias-Fischer committed:
Skip Gaussian blur tests

Tobias-Fischer committed:
non-verbose on aarch64 as limit exceeds on travis

hmaarrfk committed:
Merge pull request #70 from Tobias-Fischer/patch-1

Bump to 0.15.1 + pytorch 2.0

h-vetinari committed:
bump to 0.15.2

h-vetinari committed:
MNT: Re-rendered with conda-build 3.25.0, conda-smithy 3.23.1, and conda-forge-pinning 2023.06.02.08.08.57

github-actions[bot] committed:
torchvision 0.15.2 (#73)

automerged PR by conda-forge/automerge-action

xhochy committed:
Cross-compile aarch64

xhochy committed:
MNT: Re-rendered with conda-build 3.24.0, conda-smithy 3.23.1, and conda-forge-pinning 2023.06.13.13.09.55

xhochy committed:
Patch TORCH_LIB_PATH in setup.py

xhochy committed:
Emulated tests exceed the 6h CI threshold

xhochy committed:
MNT: Re-rendered with conda-build 3.24.0, conda-smithy 3.23.1, and conda-forge-pinning 2023.06.21.09.36.35

xhochy committed:
Skip pytest on cross-compile

xhochy committed:
MNT: Re-rendered with conda-build 3.24.0, conda-smithy 3.23.1, and conda-forge-pinning 2023.06.22.04.03.15

h-vetinari committed:
Merge pull request #74 from xhochy/cross-compile-aarch64

Cross-compile aarch64

hmaarrfk committed:
Update pin to libjpeg-turbo to 3

h-vetinari committed:
Free up some disk space before main CI

h-vetinari committed:
Skip CUDA on aarch and a spuriously failing test

h-vetinari committed:
MNT: Re-rendered with conda-build 3.26.1, conda-smithy 3.26.3, and conda-forge-pinning 2023.10.02.03.27.19

github-actions[bot] committed:
Turbo3 (#75)

automerged PR by conda-forge/automerge-action

carterbox committed:
REF: Add upstream dependency version specs

Restrict python and pytorch version ranges as specified upstream.
This is mentioned in conda-forge/torchvision-feedstock#22

carterbox committed:
NEW: Add run dependency on cuda-version

Fixes conda-forge/torchvision-feedstock#79

carterbox committed:
BLD: Bump build number

carterbox committed:
MNT: Re-rendered with conda-build 3.27.0, conda-smithy 3.28.0, and conda-forge-pinning 2023.11.08.08.48.42

carterbox committed:
REF: pin pytorch using build string instead of cuda-version package

carterbox committed:
Add site-packages to rpaths

carterbox committed:
Add missing pip args

carterbox committed:
Add missing pip args

carterbox committed:
Skip broken box drawing test

Commit:
MNT: Re-rendered with conda-build 3.27.0, conda-smithy 3.28.0, and conda-forge-pinning 2023.11.09.20.47.34

hmaarrfk committed:
Merge pull request #80 from carterbox/update-dep-pinnings

BUG: Update dep pinnings

regro-cf-autotick-bot committed:
Rebuild for pytorch21

regro-cf-autotick-bot committed:
MNT: Re-rendered with conda-build 3.27.0, conda-smithy 3.28.0, and conda-forge-pinning 2023.11.13.03.32.18

h-vetinari committed:
Merge pull request #81 from regro-cf-autotick-bot/rebuild-pytorch21-0-1_h53622f

Rebuild for pytorch21

hmaarrfk committed:
Update to 0.16.1

hmaarrfk committed:
MNT: Re-rendered with conda-build 3.27.0, conda-smithy 3.30.0, and conda-forge-pinning 2023.11.26.16.08.11

hmaarrfk committed:
Merge pull request #83 from hmaarrfk/0.16.1

Update to 0.16.1, Pytorch 2.1

regro-cf-autotick-bot committed:
Rebuild for CUDA 11.8 w/arch support

regro-cf-autotick-bot committed:
MNT: Re-rendered with conda-build 3.27.0, conda-smithy 3.30.0, and conda-forge-pinning 2023.11.27.19.54.17

hmaarrfk committed:
Add archlist for cuda

jakirkham committed:
Drop old CUDA `skip`s

As conda-forge uses CUDA 11.2+, these old CUDA `skip`s can be dropped.

github-actions[bot] committed:
Rebuild for CUDA 11.8 w/arch support (#84)

automerged PR by conda-forge/automerge-action

regro-cf-autotick-bot committed:
Rebuild for CUDA 12 w/arch + Windows support

The transition to CUDA 12 SDK includes new packages for all CUDA libraries and
build tools. Notably, the cudatoolkit package no longer exists, and packages
should depend directly on the specific CUDA libraries (libcublas, libcusolver,
etc) as needed. For an in-depth overview of the changes and to report problems
[see this issue]( https://github.com/conda-forge/conda-forge.github.io/issues/1963 ).
Please feel free to raise any issues encountered there. Thank you! :pray:

regro-cf-autotick-bot committed:
MNT: Re-rendered with conda-build 3.27.0, conda-smithy 3.30.1, and conda-forge-pinning 2023.11.28.22.13.23

weiji14 committed:
Remove if-then statements for CUDA 11.1 or below

Since conda-forge only builds CUDA11.2+ now.

weiji14 committed:
Add libcublas-dev, libcusolver-dev, libcusparse-dev to build/host deps

Needed to compile torchvision on CUDA 12.

Commit:
MNT: Re-rendered with conda-build 3.28.1, conda-smithy 3.30.1, and conda-forge-pinning 2023.12.12.14.02.26

hmaarrfk committed:
Update recipe/meta.yaml

github-actions[bot] committed:
Rebuild for CUDA 12 w/arch (#86)

automerged PR by conda-forge/automerge-action

conda-forge-webservices[bot] committed:
dummy commit for rerendering

hmaarrfk committed:
Rebuild for increased compatibility

Commit:
MNT: Re-rendered with conda-build 24.1.2, conda-smithy 3.30.4, and conda-forge-pinning 2024.02.23.21.25.19

hmaarrfk committed:
Skip two more tests

hmaarrfk committed:
Remove split recipe stuffs. Build for python 3.12

hmaarrfk committed:
MNT: Re-rendered with conda-build 24.1.2, conda-smithy 3.30.4, and conda-forge-pinning 2024.02.23.21.25.19

hmaarrfk committed:
Add back compatible pytorch

hmaarrfk committed:
Add python 3.12 migration

hmaarrfk committed:
MNT: Re-rendered with conda-build 24.1.2, conda-smithy 3.30.4, and conda-forge-pinning 2024.02.23.21.25.19

hmaarrfk committed:
Remove usage of distutils

hmaarrfk committed:
Add a patch for python 3.12

hmaarrfk committed:
Update meta.yaml

hmaarrfk committed:
Add pytest.ini to test files

hmaarrfk committed:
Ignore python 3.12 deprecation warningt pus

hmaarrfk committed:
add host dependency on libtorch

hmaarrfk committed:
MNT: Re-rendered with conda-build 24.1.2, conda-smithy 3.30.4, and conda-forge-pinning 2024.02.23.21.25.19

hmaarrfk committed:
Merge pull request #89 from conda-forge-admin/conda_forge_admin_88

Rebuild for increased compatibility and build for python 3.12

conda-forge-curator[bot] committed:
[ci skip] [skip ci] [cf admin skip] ***NO_CI*** admin migration CondaForgeAutomergeUpdate

h-vetinari committed:
renormalize patches

h-vetinari committed:
update version & URL

h-vetinari committed:
remove obsolete skip

h-vetinari committed:
rebase patches

h-vetinari committed:
apply stdlib transformations

h-vetinari committed:
add pytorch 2.3 migration

h-vetinari committed:
MNT: Re-rendered with conda-build 24.5.0, conda-smithy 3.35.1, and conda-forge-pinning 2024.05.16.03.01.34

h-vetinari committed:
add missing test dependence

h-vetinari committed:
Merge pull request #90 from h-vetinari/bump

torchvision v0.18; pytorch 2.3 migration

regro-cf-autotick-bot committed:
Rebuild for numpy 2.0

TL;DR: The way we build against numpy has changed as of numpy 2.0. This bot
PR has updated the recipe to account for the changes (see below for details).
The numpy 2.0 package itself is currently only available from a special release
channel (`conda-forge/label/numpy_rc`) and will not be available on the main
`conda-forge` channel until the release of numpy 2.0 GA.

The biggest change is that we no longer need to use the oldest available numpy
version at build time in order to support old numpy version at runtime - numpy
will by default use a compatible ABI for the oldest still-supported numpy versions.

Additionally, we no longer need to use `{{ pin_compatible("numpy") }}` as a
run requirement - this has been handled for more than two years now by a
run-export on the numpy package itself. The migrator will therefore remove
any occurrences of this.

However, by default, building against numpy 2.0 will assume that the package
is compatible with numpy 2.0, which is not necessarily the case. You should
check that the upstream package explicitly supports numpy 2.0, otherwise you
need to add a `- numpy <2` run requirement until that happens (check numpy
issue 26191 for an overview of the most important packages).

Note that the numpy release candidate promises to be ABI-compatible with the
final 2.0 release. This means that building against 2.0.0rc1 produces packages
that can be published to our main channels.

If you already want to use the numpy 2.0 release candidate yourself, you can do
```
conda config --add channels conda-forge/label/numpy_rc
```
or add this channel to your `.condarc` file directly.

### To-Dos:
  * [ ] Match run-requirements for numpy (i.e. check upstream `pyproject.toml` or however the project specifies numpy compatibility)
    * If upstream is not yet compatible with numpy 2.0, add `numpy <2` upper bound under `run:`.
    * If upstream is already compatible with numpy 2.0, nothing else should be necessary in most cases.
    * If upstream requires a minimum numpy version newer than 1.19, you can add `numpy >=x.y` under `run:`.
  * [ ] Remove any remaining occurrences of `{{ pin_compatible("numpy") }}` that the bot may have missed.

PS. If the build does not compile anymore, this is almost certainly a sign that
the upstream project is not yet ready for numpy 2.0; do not close this PR until
a version compatible with numpy 2.0 has been released upstream and on this
feedstock (in the meantime, you can keep the bot from reopening this PR in
case of git conflicts by marking it as a draft).

regro-cf-autotick-bot committed:
MNT: Re-rendered with conda-build 24.5.0, conda-smithy 3.36.0, and conda-forge-pinning 2024.05.25.13.27.35

Commit:
ENH updated version to 0.18.1

Commit:
MNT: Re-rendered with conda-build 24.5.1, conda-smithy 3.36.1, and conda-forge-pinning 2024.06.09.05.55.11

github-actions[bot] committed:
torchvision v0.18.1 (#93)

automerged PR by conda-forge/automerge-action

h-vetinari committed:
backport fix for numpy 2.0 compat

github-actions[bot] committed:
Rebuild for numpy 2.0 (#91)

automerged PR by conda-forge/automerge-action

danpetry committed:
Merge remote-tracking branch 'cf/main' into PKG-2916-0.18.1

danpetry committed:
try not to skip win, for a first pass

danpetry committed:
remove stdlib, we don't use it

danpetry committed:
Add wheel because we use it; adapt for our cudatoolkit 11.8

danpetry committed:
Reinstate pip check; add comment to check ffmpeg

danpetry committed:
Don't skip any tests, for a first pass

danpetry committed:
Some rendering and other bugfixes

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(125): `conda run conda-lint /tmp/abs_0b_equdxjl/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:69: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:71: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:59: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:44: patch_must_be_in_build: patch must be in build when source/patches is set.
===== Final Report: =====
1 Error and 3 Warnings were found

</details>

danpetry committed:
point win build system towards .lib files; whitelist the files that aren't being found, same as with pytorch; use normal jpeg as we have only got libjpeg-turbo for mac (they are api compatible)

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(125): `conda run conda-lint /tmp/abs_5a6qkz9x49/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:72: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:74: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:47: patch_must_be_in_build: patch must be in build when source/patches is set.
===== Final Report: =====
1 Error and 2 Warnings were found

</details>

danpetry committed:
whitelist for linux too; turn on debug printing for win batch

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(125): `conda run conda-lint /tmp/abs_9bjdkwayyi/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:75: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:77: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:50: patch_must_be_in_build: patch must be in build when source/patches is set.
===== Final Report: =====
1 Error and 2 Warnings were found

</details>

danpetry committed:
Fall back to 0.18.0 for compatibility with torch 2.3.0; try some more things to find libs on win; correct test syntax

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(125): `conda run conda-lint /tmp/abs_61omuubn3c/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:75: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:77: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:50: patch_must_be_in_build: patch must be in build when source/patches is set.
===== Final Report: =====
1 Error and 2 Warnings were found

</details>

danpetry committed:
disable some tests, fiddle with win configuration for test purposes

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(125): `conda run conda-lint /tmp/abs_63wa2jat4_/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:78: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:76: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:52: patch_must_be_in_build: patch must be in build when source/patches is set.
===== Final Report: =====
1 Error and 2 Warnings were found

</details>

danpetry committed:
bugfix

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(125): `conda run conda-lint /tmp/abs_7eubp_5lt5/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:78: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:76: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:52: patch_must_be_in_build: patch must be in build when source/patches is set.
===== Final Report: =====
1 Error and 2 Warnings were found

</details>

danpetry committed:
Bugfix again

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(125): `conda run conda-lint /tmp/abs_54xd4gx8_r/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:78: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:76: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:52: patch_must_be_in_build: patch must be in build when source/patches is set.
===== Final Report: =====
1 Error and 2 Warnings were found

</details>

danpetry committed:
Simplify tests, reinstate ffmpeg

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(125): `conda run conda-lint /tmp/abs_65wf7kwe7l/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:75: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:69: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:73: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:52: patch_must_be_in_build: patch must be in build when source/patches is set.
===== Final Report: =====
1 Error and 3 Warnings were found

</details>

danpetry committed:
skip s390x

danpetry committed:
Use libjpeg-turbo (from the staging channel for now)

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(125): `conda run conda-lint /tmp/abs_54rd2s5fqu/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:69: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:67: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:73: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:75: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:52: patch_must_be_in_build: patch must be in build when source/patches is set.
===== Final Report: =====
1 Error and 4 Warnings were found

</details>

danpetry committed:
Remove unused patch; remove smoke test for libjpeg-turbo usage (for reasons described in patch)

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(125): `conda run conda-lint /tmp/abs_fdkrgqg8_y/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:74: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:76: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:70: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:53: patch_must_be_in_build: patch must be in build when source/patches is set.
===== Final Report: =====
1 Error and 3 Warnings were found

</details>

danpetry committed:
fixup

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(125): `conda run conda-lint /tmp/abs_78ap2o432p/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:70: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:76: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:74: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:53: patch_must_be_in_build: patch must be in build when source/patches is set.
===== Final Report: =====
1 Error and 3 Warnings were found

</details>

danpetry committed:
Add test artifacts

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(125): `conda run conda-lint /tmp/abs_98lvd9r8ri/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:76: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:74: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:70: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:53: patch_must_be_in_build: patch must be in build when source/patches is set.
===== Final Report: =====
1 Error and 3 Warnings were found

</details>

danpetry committed:
skip cuda and python 3.12, selector bugfix for pytorch

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(125): `conda run conda-lint /tmp/abs_61l1vnr2xu/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:72: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:76: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:78: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:55: patch_must_be_in_build: patch must be in build when source/patches is set.
===== Final Report: =====
1 Error and 3 Warnings were found

</details>

danpetry committed:
Correct DSO checks

danpetry committed:
Correct erroneous selectors

danpetry committed:
more version selectors corrections

danpetry committed:
Correct selectors

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(125): `conda run conda-lint /tmp/abs_cehq4mecz5/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:80: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:78: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:74: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:57: patch_must_be_in_build: patch must be in build when source/patches is set.
===== Final Report: =====
1 Error and 3 Warnings were found

</details>

danpetry committed:
provide ptx for older archs to give better compatibility

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(125): `conda run conda-lint /tmp/abs_315zayo810/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:74: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:78: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.
- clone/recipe/meta.yaml:80: host_section_needs_exact_pinnings: Linked libraries host should have exact version pinnings.

===== ERRORS =====
- clone/recipe/meta.yaml:57: patch_must_be_in_build: patch must be in build when source/patches is set.
===== Final Report: =====
1 Error and 3 Warnings were found

</details>

danpetry commented on abs.yaml:
```diff
@@ -0,0 +1,2 @@
+channels:
+  - https://staging.continuum.io/prefect/fs/libjpeg-turbo-feedstock/pr1/8da19a2
```
abs.yaml needs removing

JeanChristopheMorinPerso commented on recipe/meta.yaml:
```diff
@@ -1,119 +1,121 @@
-    # see https://github.com/pytorch/vision/issues/5419#issuecomment-1193483864
-    # and https://github.com/conda-forge/torchvision-feedstock/commit/4ec4b5f4eb4889dbb1f8da34662ea622bb4b3828
-    # - ffmpeg
+    - ffmpeg
```
We should probably pin ffmpeg?

danpetry replied:
yep. waiting for the new one.

danpetry replied:
pinned and added staging channel

JeanChristopheMorinPerso commented on recipe/meta.yaml:
```diff
@@ -1,119 +1,121 @@
-{% set version = "0.15.2" %}
+{% set version = "0.18.0" %}
```
Should we bump it to 0.18.1?

cbouss replied:
That's the diff between both versions, so it looks like yes:
https://github.com/pytorch/vision/compare/v0.18.0...v0.18.1

danpetry replied:
apparently, v0.18.1 is incompatible with pytorch 2.3.0. It's an update for v2.3.1. According to the pytorch maintainers.

danpetry replied:
oh, I see. alright yeah.

danpetry replied:
updated

JeanChristopheMorinPerso commented on recipe/meta.yaml:
```diff
@@ -1,119 +1,121 @@
+    - ffmpeg
+    - numpy
+    # exclude 8.3.0 and 8.3.1 specifically due to pytorch/vision#4146, python-pillow/Pillow#5571
+    - pillow >=5.3.0,!=8.3.0,!=8.3.1
```
I don't think it's needed at build time. Pillow doesn't ship any header files.

skupr-anaconda replied:
It's a runtime dependency https://github.com/pytorch/vision/blob/6043bc250768b129e90a5321e318c1d51ee48a5c/setup.py#L68

danpetry replied:
removed

JeanChristopheMorinPerso commented on recipe/meta.yaml:
```diff
@@ -1,119 +1,121 @@
+    - pillow >=5.3.0,!=8.3.0,!=8.3.1
+    #- libtorch
+    - pytorch ={{ compatible_pytorch }}.*=*{{ torch_proc_type }}*
+    - requests
```
Is requests really needed at build time?

danpetry replied:
removed

JeanChristopheMorinPerso commented on recipe/meta.yaml:
```diff
@@ -1,119 +1,121 @@
+    - pip check
+    - python test/smoke_test.py
+    # Vast majority of tests pass. Failures can be seen in logs. Tests crash on Windows due to a memoryerror.
+    - pytest --verbose --durations=50 test/ || true    # [not win]
```
Can we be more explicit about which tests fail? Edit: Some tests download stuff. We should disable these (`test_datasets_download.py`).

cbouss replied:
Agreed, I'd like to see the or true removed. Apart from linux-aarch64 and win-64, all platforms all platforms have a limited number of failures.
`test_datasets_download` (ignored in pytest.ini), `test_detection_model` (there are warnings in the logs that make me think theses tests may be flaky), `test_transforms` (based on a former comment on this recipe: torchvision has bugs with certain ffmpeg versions)

danpetry replied:
thanks, halfway through this. maybe some tests will pass with a newer ffmpeg.

JeanChristopheMorinPerso commented on recipe/patches/0003-dont-test-for-turbo-jpeg.patch:
```diff
@@ -0,0 +1,19 @@
+Using libjpeg-turbo here would require that we update the whole distro to use
```
```suggestion
Using libjpeg-turbo here would require that we update defaults to use
```
To be more clear.

JeanChristopheMorinPerso commented:
On `osx-64m`, `osx-arm64` and `win-64`, I see
```
2024-07-12T23:05:14.751145+00:00  | INFO     |   WARNING (torchvision): run-exports library package defaults/osx-64::ffmpeg==4.2.2=h97e5cf8_0 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want
 then add it to `build/ignore_run_exports`)
```
That seems suspicious to me.

Additionally, on win-64, jpeg seems to be statically linked. Edit: I created a PR in jpeg to fix this: https://github.com/AnacondaRecipes/jpeg-feedstock/pull/6.

skupr-anaconda commented on recipe/meta.yaml:
```diff
@@ -1,119 +1,121 @@
-    # and https://github.com/conda-forge/torchvision-feedstock/commit/4ec4b5f4eb4889dbb1f8da34662ea622bb4b3828
-    # - ffmpeg
+    - ffmpeg
+    - numpy
```
numpy is needed only at runtime, right? See https://github.com/pytorch/vision/blob/v0.18.0/pyproject.toml#L18 and https://github.com/pytorch/vision/blob/v0.18.0/setup.py#L60-L63

danpetry replied:
done

skupr-anaconda commented on recipe/meta.yaml:
```diff
@@ -1,119 +1,121 @@
   run:
-    # GPU requirements
-    - {{ pin_compatible('cudatoolkit', max_pin='x.x') }}  # [pytorch_variant == "gpu"]
-    - {{ pin_compatible('numpy') }}
```
`numpy` is missing?

danpetry replied:
done

cbouss commented on recipe/meta.yaml:
```diff
@@ -1,119 +1,121 @@
+    # seems to be necessary for aarch???
+    - patches/0001-Monkeypatch-TORCH_LIB_PATH.patch
+    # backport https://github.com/pytorch/vision/pull/8463 for numpy 2.0 compat
+    - patches/0002-Make-adjust_hue-work-with-numpy-2.0-8463.patch
```
This seems to be the code that is failing on linux-aarch64.
Can you try not applying that patch?
This can be revisited once we attempt a numpy2 build.

danpetry replied:
in progress

cbouss commented on recipe/meta.yaml:
```diff
@@ -1,119 +1,121 @@
-    # see https://github.com/pytorch/vision/issues/5419#issuecomment-1193483864
-    # and https://github.com/conda-forge/torchvision-feedstock/commit/4ec4b5f4eb4889dbb1f8da34662ea622bb4b3828
-    # - ffmpeg
+    - ffmpeg
```
```suggestion
    - ffmpeg 4.2
```

danpetry replied:
done, actually pinned to 6.1.1

cbouss commented on recipe/meta.yaml:
```diff
@@ -1,119 +1,121 @@
-    # and https://github.com/conda-forge/torchvision-feedstock/commit/4ec4b5f4eb4889dbb1f8da34662ea622bb4b3828
-    # - ffmpeg
+    - ffmpeg
+    - numpy
```
```suggestion
```
Agreed with Serhii

danpetry replied:
done

cbouss commented on recipe/meta.yaml:
```diff
@@ -1,119 +1,121 @@
+    - ffmpeg
+    - numpy
+    # exclude 8.3.0 and 8.3.1 specifically due to pytorch/vision#4146, python-pillow/Pillow#5571
+    - pillow >=5.3.0,!=8.3.0,!=8.3.1
```
```suggestion
```

danpetry replied:
done

cbouss commented on recipe/meta.yaml:
```diff
@@ -1,119 +1,121 @@
+    - pillow >=5.3.0,!=8.3.0,!=8.3.1
+    #- libtorch
+    - pytorch ={{ compatible_pytorch }}.*=*{{ torch_proc_type }}*
+    - requests
```
```suggestion
```

danpetry replied:
done

cbouss commented on recipe/meta.yaml:
```diff
@@ -1,119 +1,121 @@
+    - cudatoolkit {{ cuda_compiler_version }}    # [cuda_compiler_version == "11.8"]
+    - cuda-version {{ cuda_compiler_version }}   # [cuda_compiler_version != "None"]
+    - pillow >=5.3.0,!=8.3.0,!=8.3.1
     - requests
```
```suggestion
```
I don't see this being required.

cbouss commented on recipe/meta.yaml:
```diff
@@ -1,119 +1,121 @@
+    - {{ pin_compatible('cudnn') }}           # [cuda_compiler_version != "None"]
+    - cudatoolkit {{ cuda_compiler_version }}    # [cuda_compiler_version == "11.8"]
+    - cuda-version {{ cuda_compiler_version }}   # [cuda_compiler_version != "None"]
+    - pillow >=5.3.0,!=8.3.0,!=8.3.1
```
```suggestion
    - pillow >=5.3.0,!=8.3.0,!=8.3.1
    - numpy <2.0a0
```
If you end up removing the numpy2 compat patch. Otherwise just numpy.

danpetry replied:
in progress

cbouss commented on recipe/meta.yaml:
```diff
@@ -1,119 +1,121 @@
+    - pytest-mock
+  source_files:
+    - test
+    - gallery
```
```suggestion
    - gallery
    - pytest.ini
```
Should solve some test failures by using the ignore patterns from upstream.
https://github.com/pytorch/vision/blob/v0.18.1/pytest.ini

danpetry replied:
done

cbouss commented on recipe/meta.yaml:
```diff
@@ -1,119 +1,121 @@
 test:
+  requires:
+    - pip
+    - pytest
```
```suggestion
    - pytest <8
```
Because of https://github.com/pytorch/vision/issues/8238

danpetry replied:
done

danpetry committed:
Address some review comments

 - Update to v0.18.1 (latest)
 - pin ffmpeg to latest version
 - explicitly skip some tests (halfway done with this before can remove or-true)
 - remove unneeded host deps

danpetry committed:
Get new ffmpeg from staging channel

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(125): `conda run conda-lint /tmp/abs_cafndakq5o/clone` failed. (See above for error)
The following problems have been found:

===== ERRORS =====
- clone/recipe/meta.yaml:57: patch_must_be_in_build: patch must be in build when source/patches is set.
===== Final Report: =====
1 Error and 0 Warnings were found

</details>

danpetry committed:
Remove hue adjustment patch

danpetry committed:
correct deps

danpetry committed:
get ffmpeg from staging channel really

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(125): `conda run conda-lint /tmp/abs_59kma4fzjq/clone` failed. (See above for error)
The following problems have been found:

===== ERRORS =====
- clone/recipe/meta.yaml:57: patch_must_be_in_build: patch must be in build when source/patches is set.
===== Final Report: =====
1 Error and 0 Warnings were found

</details>

danpetry committed:
Use updated jpeg on win; don't patch linux-aarch64, I think it's for cross-compilation; pin cudnn

danpetry commented on recipe/meta.yaml:
```diff
@@ -1,119 +1,129 @@
-{% set tests_to_skip = tests_to_skip + " or test_detection_model" %}
-{% set tests_to_skip = tests_to_skip + " or test_quantized_classification_model" %}  # [linux]
+    - pytorch ={{ compatible_pytorch }}.*=*{{ torch_proc_type }}*
+    - {{ pin_compatible('cudnn') }}              # [cuda_compiler_version != "None"]
```
There's an overdepending warning for this. Should be investigated.

danpetry committed:
Remove cudnn, it's not used

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(125): `conda run conda-lint /tmp/abs_695ykai4qt/clone` failed. (See above for error)
The following problems have been found:

===== ERRORS =====
- clone/recipe/meta.yaml:57: patch_must_be_in_build: patch must be in build when source/patches is set.
===== Final Report: =====
1 Error and 0 Warnings were found

</details>

danpetry committed:
skip more tests, use updated jpeg

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(125): `conda run conda-lint /tmp/abs_b4xvqh9soo/clone` failed. (See above for error)
The following problems have been found:

===== ERRORS =====
- clone/recipe/meta.yaml:57: patch_must_be_in_build: patch must be in build when source/patches is set.
===== Final Report: =====
1 Error and 0 Warnings were found

</details>

danpetry committed:
Try reducing tests on linux again to prevent resource exhaustion

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(125): `conda run conda-lint /tmp/abs_e06vvh8nd0/clone` failed. (See above for error)
The following problems have been found:

===== ERRORS =====
- clone/recipe/meta.yaml:57: patch_must_be_in_build: patch must be in build when source/patches is set.
===== Final Report: =====
1 Error and 0 Warnings were found

</details>

danpetry committed:
Reduce test set; remove unused patches

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(125): `conda run conda-lint /tmp/abs_d3vwqn683b/clone` failed. (See above for error)
The following problems have been found:

===== ERRORS =====
- clone/recipe/meta.yaml:53: patch_must_be_in_build: patch must be in build when source/patches is set.
===== Final Report: =====
1 Error and 0 Warnings were found

</details>

danpetry commented:
[log for cuda build latest commit](https://github.com/user-attachments/files/16461227/3evenlesstests.log)

danpetry commented:
@cbouss @JeanChristopheMorinPerso @skupr-anaconda could I get a re-review please?

skupr-anaconda reviewed (approved):
LGTM

cbouss commented on abs.yaml:
```diff
@@ -0,0 +1,2 @@
+channels:
+  - https://staging.continuum.io/prefect/fs/jpeg-feedstock/pr7/3f0a23b
```
```suggestion
  - https://staging.continuum.io/prefect/fs/jpeg-feedstock/pr7/0c4591b
```
3f0a23b is gone, so I am not sure which version of jpeg it was.

danpetry replied:
updated jpeg is uploaded now so I deleted abs.yaml

cbouss reviewed (commented):
Can you re-trigger a build with updated jpeg?
The recipe looks good, and you did great work there. I have yet to review the logs. (The prefect logs have expired already.)

danpetry committed:
Jpeg uploaded now

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(125): `conda run conda-lint /tmp/abs_2cc8amsx8y/clone` failed. (See above for error)
The following problems have been found:

===== ERRORS =====
- clone/recipe/meta.yaml:53: patch_must_be_in_build: patch must be in build when source/patches is set.
===== Final Report: =====
1 Error and 0 Warnings were found

</details>

danpetry commented:
@cbouss rebuilt with new jpeg

cbouss reviewed (approved):
LGTM 🎉

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(125): `conda run conda-lint /tmp/abs_7ahgefciqg/clone` failed. (See above for error)
The following problems have been found:

===== ERRORS =====
- clone/recipe/meta.yaml:53: patch_must_be_in_build: patch must be in build when source/patches is set.
===== Final Report: =====
1 Error and 0 Warnings were found

</details>
